client/__tests__/quizPage.test.jsx

```diff
@@ -0,0 +1,89 @@
+import TEST_QUESTION from "./testdata/TEST_QUESTION.json";
+import ReactDOM from "react-dom";
+import React from "react";
+import { act } from "react-dom/test-utils";
+import { Simulate } from "react-dom/test-utils";
+import { MemoryRouter } from "react-router-dom";
+import { QuizPage } from "../components/quizPage";
+
+async function getQuestion() {
+  return {
+    id: TEST_QUESTION.id,
+    category: TEST_QUESTION.category,
+    question: TEST_QUESTION.question,
+    answers: TEST_QUESTION.answers,
+  };
+}
+async function checkAnswer(id, answer) {
+  if (answer === "answer_b") {
+    return { isCorrect: true };
+  } else {
+    return { isCorrect: false };
+  }
+}
+
+const defaultQuestionApi = {
+  getQuestion,
+  checkAnswer: jest.fn().mockImplementation(checkAnswer),
+};
+
+describe("question", () => {
+  it("renders a question", async () => {
+    const element = document.createElement("div");
+    await act(async () => {
+      await ReactDOM.render(
+        <MemoryRouter>
+          <QuizPage questionApi={defaultQuestionApi} />
+        </MemoryRouter>,
+        element
+      );
+    });
+    expect(element).toMatchSnapshot();
```

Comment:
Bra eksempel på snapshot test. Om dere har lyst kan dere bruke `pretty` for å formattere snapshotte. Det gir bedre differ. Det har ingenting å si på eksamen, men kan være til hjelp under utviklingen